src/data-workspace/data-workspace.js

```diff
@@ -46,6 +47,20 @@ export const DataWorkspace = ({ selectionHasNoFormMessage }) => {
     // used to reset form-state when context-selection is changed
     const formKey = useContextSelectionId()
 
+    // force refetch when context-selection changes
+    useEffect(() => {
+        if (isValidSelection) {
+            // note this will only refetch "active"/mounted queries,
+            // so it's safe to not provide a queryKey
+            queryClient.invalidateQueries(null, {
+                // if new selection is not in cache, a fetch will already be in-flight
+                // so we do not need to send another one.
+                cancelRefetch: false,
+            })
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [formKey])
+
```

Comment:
I don't like this rule to be disabled, since I have actually seen bugs with stale values in the `useEffect` callback due to this. I do see the problem though. As a solution, you could do the following:

Add this below line 48:
```js
const validFormKey = isValidSelection ? formKey : ''
```

On line 52, change change the check on `if (isValidSelection)` to a check for `if (validFormKey)`

On line 62, add the `validFormKey` to the dependency array. And also add `queryClient`, this should be a stable reference, so that shouldn't matter.